js/main.js

```diff
@@ -4,30 +4,83 @@ var $recipePhoto = document.querySelector('#recipe-photo');
 var $ingredientList = document.querySelector('#ingredient-list');
 var $instructionList = document.querySelector('#instruction-list');
 var $savedRecipesView = document.querySelector('#saved-recipes');
+var $searchRecipesLink = document.querySelector('#search-recipes-link');
 var $savedRecipesLink = document.querySelector('#saved-recipes-link');
 var $views = document.querySelectorAll('.view');
 var $readRecipeTitle = document.querySelector('#read-recipe-title');
 var $readRecipePhoto = document.querySelector('#read-recipe-photo');
 var $readRecipeIngredientList = document.querySelector('#read-recipe-ingredient-list');
 var $readRecipeInstructionList = document.querySelector('#read-recipe-instruction-list');
 var $listCol = document.querySelector('#list-col');
+var $searchForm = document.querySelector('#search-form');
+var $searchHeader = document.querySelector('#search-header');
+var $searchList = document.querySelector('#search-list');
 var currentlyEditing = false;
 var currentRecipe;
+var results;
+
+var recipeRequest = new XMLHttpRequest();
 
 populateSavedRecipesView();
 
+$searchForm.elements['search-button'].addEventListener('click', searchRecipes);
+recipeRequest.addEventListener('load', getAllResultIngredients);
+
 $listCol.addEventListener('click', deleteItem);
 $listCol.addEventListener('click', saveItemEdit);
 $listCol.addEventListener('click', openItemEditor);
 $savedRecipesView.addEventListener('click', openEditor);
 $savedRecipesView.addEventListener('click', readRecipe);
 $savedRecipesLink.addEventListener('click', swapView);
+$searchRecipesLink.addEventListener('click', swapView);
 $newRecipeForm.elements['save-button'].addEventListener('click', saveRecipe);
 $newRecipeForm.elements['add-instruction-button'].addEventListener('click', addInstruction);
 $newRecipeForm.elements['add-ingredient-button'].addEventListener('click', addIngredient);
 $newRecipeForm.elements['photo-url'].addEventListener('input', updateRecipePhoto);
 $newRecipeForm.elements.title.addEventListener('input', updateRecipeTitle);
 
+function storeIngredients(event) {
+  var ingredients = JSON.parse(event.target.response);
```

Comment:
Is this necessary? Is the data not already arriving parsed? If so, you may need to set the response type: https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest/responseType